Razor build server (process 27970) failed to shut down: The shutdown command failed: The application to execute does not exist

## Steps to reproduce
`$ dotnet build-server shutdown` error:
```
Razor build server (process 27970) failed to shut down: The shutdown command failed: The application to execute does not exist: '/usr/share/dotnet/sdk/NuGetFallbackFolder/microsoft.aspnetcore.razor.design/2.2.0/tools/rzc.dll'
```
Why does it look for 2.2.0 when I am using 3.0?
## Expected  behavior
Shutdown properly without error

## Actual behavior


## Environment data
`dotnet --info` output:
```
$ dn --info
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100
 Commit:    04339c3a26

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  19.04
 OS Platform: Linux
 RID:         ubuntu.19.04-x64
 Base Path:   /usr/share/dotnet-3.0.100/sdk/3.0.100/

Host (useful for support):
  Version: 3.0.0
  Commit:  7d57652f33

.NET Core SDKs installed:
  3.0.100 [/usr/share/dotnet-3.0.100/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.0.0 [/usr/share/dotnet-3.0.100/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.0.0 [/usr/share/dotnet-3.0.100/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

I think this issue just came back. Our perf build ran into the same issue with .NET 5. We use the command ```dotnet build-server shutdown``` to set up before each iteration and recently this happens very frequently with the master branch. @wli3 
```
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO] ----------------------------------------------
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO] Initializing logger 2020-06-02 20:37:30.187270
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO] ----------------------------------------------
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO] $ pushd "/home/helixbot/work/A4AE08C9/p"
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO] $ dotnet build-server shutdown
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO] Shutting down MSBuild server...
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO] Shutting down VB/C# compiler server...
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO] Shutting down Razor build server (process 13722)...
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO] VB/C# compiler server shut down successfully.
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO] Razor build server (process 13722) failed to shut down: The shutdown command failed: The application to execute does not exist: '/home/helixbot/work/AEFA0996/w/AD220924/e/.dotnet14993/sdk/5.0.100-preview.6.20266.3/Sdks/Microsoft.NET.Sdk.Razor/tools/netcoreapp3.0/rzc.dll'
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:30][INFO]
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:31][INFO] MSBuild server shut down successfully.
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:31][INFO] $ popd
[2020/06/02 20:37:31][INFO] [2020/06/02 20:37:31][ERROR] Process exited with status 1
```

--- a/src/Cli/dotnet/BuildServer/RazorServer.cs
+++ b/src/Cli/dotnet/BuildServer/RazorServer.cs
@@ -36,6 +36,14 @@ namespace Microsoft.DotNet.BuildServer
 
         public void Shutdown()
         {
+            if(!_fileSystem.File.Exists(PidFile.ServerPath.Value))
+            {
+                // The razor server path doesn't exist anymore so trying to shut it down would fail
+                // Ensure the pid file is cleaned up so we don't try to shut it down again
+                DeletePidFile();
+                return;
+            }
+
             var command = _commandFactory
                 .Create(
                     "exec",
@@ -60,6 +68,11 @@ namespace Microsoft.DotNet.BuildServer
 
             // After a successful shutdown, ensure the pid file is deleted
             // If the pid file was left behind due to a rude exit, this ensures we don't try to shut it down again
+            DeletePidFile();
+        }
+
+        void DeletePidFile()
+        {
             try
             {
                 if (_fileSystem.File.Exists(PidFile.Path.Value))

--- a/src/Tests/dotnet.Tests/BuildServerTests/RazorServerTests.cs
+++ b/src/Tests/dotnet.Tests/BuildServerTests/RazorServerTests.cs
@@ -2,6 +2,7 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using FluentAssertions;
@@ -31,14 +32,16 @@ namespace Microsoft.DotNet.Tests.BuildServerTests
             string pidDirectory = Path.GetFullPath("var/pids/build");
             string pidFilePath = Path.Combine(pidDirectory, $"{RazorPidFile.FilePrefix}{ProcessId}");
 
+            var serverPath = Path.GetFullPath("path/to/rzc.dll");
+
             var fileSystemMock = new FileSystemMockBuilder()
                 .AddFile(pidFilePath, "")
+                .AddFile(serverPath, "")
                 .UseCurrentSystemTemporaryDirectory()
                 .Build();
 
             fileSystemMock.File.Exists(pidFilePath).Should().BeTrue();
-
-            var serverPath = Path.Combine(fileSystemMock.Directory.CreateTemporaryDirectory().DirectoryPath, "path/to/rzc.dll");
+            fileSystemMock.File.Exists(serverPath).Should().BeTrue();
 
             var server = new RazorServer(
                 pidFile: new RazorPidFile(
@@ -68,14 +71,16 @@ namespace Microsoft.DotNet.Tests.BuildServerTests
             string pidDirectory = Path.GetFullPath("var/pids/build");
             string pidFilePath = Path.Combine(pidDirectory, $"{RazorPidFile.FilePrefix}{ProcessId}");
 
+            var serverPath = Path.GetFullPath("path/to/rzc.dll");
+
             var fileSystemMock = new FileSystemMockBuilder()
                 .AddFile(pidFilePath, "")
+                .AddFile(serverPath, "")
                 .UseCurrentSystemTemporaryDirectory()
                 .Build();
 
             fileSystemMock.File.Exists(pidFilePath).Should().BeTrue();
-
-            var serverPath = Path.Combine(fileSystemMock.Directory.CreateTemporaryDirectory().DirectoryPath, "path/to/rzc.dll");
+            fileSystemMock.File.Exists(serverPath).Should().BeTrue();
 
             var server = new RazorServer(
                 pidFile: new RazorPidFile(
@@ -87,6 +92,43 @@ namespace Microsoft.DotNet.Tests.BuildServerTests
                 fileSystem: fileSystemMock);
 
             server.Shutdown();
+
+            fileSystemMock.File.Exists(pidFilePath).Should().BeFalse();
+        }
+
+        [Fact]
+        public void GivenANonExistingRazorServerPathItDeletesPidFileAndDoesNotThrow()
+        {
+            const int ProcessId = 1234;
+            const string PipeName = "some-pipe-name";
+
+            string pidDirectory = Path.GetFullPath("var/pids/build");
+            string pidFilePath = Path.Combine(pidDirectory, $"{RazorPidFile.FilePrefix}{ProcessId}");
+
+            var serverPath = Path.GetFullPath("path/to/rzc.dll");
+
+            var fileSystemMock = new FileSystemMockBuilder()
+                .AddFile(pidFilePath, "")
+                .UseCurrentSystemTemporaryDirectory()
+                .Build();
+
+            fileSystemMock.File.Exists(pidFilePath).Should().BeTrue();
+            fileSystemMock.File.Exists(serverPath).Should().BeFalse();
+
+            var commandFactoryMock = CreateCommandFactoryMock(serverPath, PipeName);
+            var server = new RazorServer(
+                pidFile: new RazorPidFile(
+                    path: new FilePath(pidFilePath),
+                    processId: ProcessId,
+                    serverPath: new FilePath(serverPath),
+                    pipeName: PipeName),
+                commandFactory: commandFactoryMock.Object,
+                fileSystem: fileSystemMock);
+
+            Action a = () => server.Shutdown();
+
+            a.Should().NotThrow();
+            commandFactoryMock.Verify(c => c.Create(It.IsAny<string>(), It.IsAny<IEnumerable<string>>(), It.IsAny<NuGetFramework>(), It.IsAny<string>()), Times.Never);
 
             fileSystemMock.File.Exists(pidFilePath).Should().BeFalse();
         }

